[Damien Koh] iP

Ssatu opened the pull request:
##Duke
> Duke duke duke duke
Who is [Duke](https://www.oracle.com/java/duke/)?
~~Why is Duke?~~

What can Duke do for _you_?
* Manage your to-do list
* Keep you informed of your deadlines
* Help you find information about your tasks with ease

Where can you use Duke? 👍 
1. On the go,
2. at home
3. basically **anywhere**!

What are some of the tasks that you can schedule?
- [x] To-do
- [x] Deadlines
- [x] Events
- [ ] Reminders


If you're a java programmer you can use it to practice Java too! Here's the `main` method:
```java
public static void main(String[] args) {
        new Duke("data.txt").run();
    }
```

Commit:
Add Gradle support

Ssatu committed:
Completed Level-1

Ssatu committed:
Level 2 complete

Ssatu committed:
Level 3 complete

Ssatu committed:
Completed level 4

Ssatu committed:
Added testing

Ssatu committed:
Completed level 6 and added more exception handling to mark and unmark.

Ssatu committed:
Added tag for level 5, fixed extra blank lines for test files.

Ssatu committed:
Adding updates to Level 7 with some changes to addTask.

Ssatu committed:
Add Level 8 changes

Ssatu committed:
Merge branch 'branch-Level-7'

Ssatu committed:
Fix differences between branches

Ssatu committed:
Merge pull request #1 from nus-cs2103-AY2223S1/add-gradle-support

Add Gradle support

Ssatu committed:
Merge remote-tracking branch 'origin/master'

Ssatu committed:
Added levels up till JUnit tests

Ssatu committed:
Added A-JavaDoc

Ssatu committed:
Adding coding standard

Ssatu committed:
Adding Level-9

Ssatu committed:
Merge branch 'branch-A-CodingStandard'

Ssatu committed:
Merge branch 'A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Task.java
#	src/main/java/duke/TaskList.java

Ssatu committed:
Fixed tests

Ssatu committed:
Stopped tracking unecessary files

Ssatu committed:
Added files

Ssatu committed:
Added checkstyle config files and made checkstyle edits

Ssatu committed:
Added Level 10 GUI updates

Ssatu committed:
Merging A-CheckStyle and Level 10 GUI updates

Ssatu committed:
Checkstyle updates

Tex-Tang commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,84 @@
+     * @param tasks The TaskList being used.
+     * @param storage The storage object being used.
+     */
+    public static String parse(String input, TaskList tasks, Storage storage) {
```
Should the `parse` method do only parsing command instead of calling the command?

Tex-Tang commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,84 @@
+            } catch (DukeException ex) {
+                return ex.toString();
+            }
+        } else if (first.equals("unmark")) {
+            //Mark a task as not done
+            try {
+                String message = tasks.unmarkTask(rest);
+                storage.writeToFile(tasks);
+                return message;
+            } catch (DukeException ex) {
+                return ex.toString();
+            }
+        } else if (first.equals("delete")) {
+            //Delete a task at a given index
+            try {
+                String message = tasks.deleteTask(rest);
+                storage.writeToFile(tasks);
+                return message;
+            } catch (DukeException ex) {
+                return ex.toString();
+            }
+        } else if (first.equals("todo") || first.equals("deadline") || first.equals("event")) {
+            //Else, add task to list
+            try {
+                String message = tasks.addTask(rest, first, false, false);
+                storage.writeToFile(tasks);
+                return message;
+            } catch (DukeException ex) {
+                return ex.toString();
+            }
+        } else if (first.equals("find")) {
+            try {
+                return tasks.find(rest);
+            } catch (DukeException ex) {
+                return ex.toString();
+            }
+        } else {
+            return "OOPS!!! I'm sorry, but I don't know what that means :-(";
+        }
+    }
```
Should we wrap these code into single `try-catch` to avoid repeating `try-catch` for each command.

Tex-Tang commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,225 @@
+        for (int i = 0; i < tasks.size(); i++) {
+            if (tasks.get(i).getTitle().contains(input)) {
+                found += 1;
+                //Add index
+                list += (found) + ".";
+                //Add task
+                list += tasks.get(i);
+                //Add nextline
+                list += "\n";
```
```suggestion
                list += String.printf("%d. %s\n", found, tasks.get(i));
```
Might be a better choice as it is more simpler.

Tex-Tang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,180 @@
+     * Runs the duke program and parses user's input.
+     */
+    public void run() {
+        Scanner scn = new Scanner(System.in);
```
Should we change `scn` to `scanner` for a clearer name?

Tex-Tang commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,225 @@
+    /**
+     * This method adds a task to the given task-list.
+     *
+     * @param noType The input String with type removed.
+     * @param type The type of the task
+     */
+    public String addTask(String noType, String type, boolean done, boolean load)
```
Should we add comment for `done` and `load` also?

Tex-Tang reviewed (commented):
Overall it is very good. Code in `Event.java`, `Deadline.java` and `ToDo.java` are very clean. Just need to fix some minor issues.

ReubenChay commented on src/main/java/duke/ToDo.java:
```diff
@@ -0,0 +1,16 @@
+ */
+public class ToDo extends Task {
+
+    public ToDo(String title, boolean done) {
```
boolean variable can start with isDone instead

ReubenChay commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+/**
+ * A task with a specified deadline.
+ */
+public class Deadline extends Task {
```
can't find much coding standard issues not already found by other reviewer. Really clear code with good use of OOP!

Ssatu committed:
Added assert statement

Ssatu committed:
Code quality changes

Ssatu committed:
Merge pull request #2 from Ssatu/branch-A-CodeQuality

Code quality changes

Ssatu committed:
Merge pull request #3 from Ssatu/branch-A-Assertions

Added assert statement

Ssatu committed:
Adding the undo() method

The undo method alllows the user to undo any commands that result in changes to the data stored.

Modified the parse() method

The parse method required an update to recieve the undo command.

Ssatu committed:
Merge pull request #5 from Ssatu/branch-A-Assertions

Adding the undo() method

Ssatu committed:
Modify functionality of undo.

Undo now redo-es and undo if called twice, allowing for greater fault tolerance in user behaviour.

Improve in the future to allow for undo or redo of multiple commands in history.

Ssatu committed:
Added User guide and some minor crediting and code style changes.

Ssatu committed:
Merge pull request #6 from Ssatu/heads/ip/master

Heads/ip/master

Ssatu committed:
Bug fixes to deadline and event commands.

Ssatu committed:
Merge pull request #7 from Ssatu/ip/master

Fix bugs in deadline and event commands.

Ssatu committed:
Fix checkstyle errors.

Ssatu committed:
Merge pull request #8 from Ssatu/ip/master

Fix checkstyle errors.

Ssatu committed:
Merge remote-tracking branch 'origin/master' into ip/master

Ssatu committed:
Add user guide details

Ssatu committed:
Update UG

Ssatu committed:
Fix UG details for mark and unmark

Ssatu committed:
Merge pull request #10 from Ssatu/ip/master

Fix UG details for mark and unmark

Ssatu committed:
Fix further UG bugs

Ssatu committed:
Merge pull request #11 from Ssatu/ip/master

Fix further UG bugs
